File: app/src/main/java/com/example/ahoyandroidtaskmudassar/utils/NotificationHelper.java
package com.example.ahoyandroidtaskmudassar.utils;

import android.annotation.TargetApi;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.ContentResolver;
import android.content.Context;
import android.content.ContextWrapper;
import android.content.Intent;
import android.media.Ringtone;
import android.media.RingtoneManager;
import android.net.Uri;
import android.os.Build;
import android.os.VibrationEffect;
import android.os.Vibrator;

import androidx.core.app.NotificationCompat;

import com.example.ahoyandroidtaskmudassar.R;

public class NotificationHelper extends ContextWrapper {
    public static final String channelID = "channelID";
    public static final String channelName = "Channel Name";
    Uri soundUri = Uri.parse(ContentResolver.SCHEME_ANDROID_RESOURCE + "://" + getApplicationContext().getPackageName() + "/" + R.raw.old_telephone);
    public static Ringtone r = null;
    public static Ringtone ringtone = null;
    public static Vibrator v = null;
    NotificationCompat.Action actionYes;

    private String name = "";
    private String temp = "";
    private int id = 0;
    private NotificationManager mManager;
    Uri sound = null;

    public NotificationHelper(Context base, String name,String temp, int Id) {
        super(base);
        this.name = name;
        this.temp = temp;
        this.id = Id;
        sound = Uri.parse(ContentResolver.SCHEME_ANDROID_RESOURCE + "://" + base.getPackageName() + "/" + R.raw.old_telephone);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            createChannel();
        }
        Intent yesIntent = new Intent(this, RingtonePlayingService.class);
        yesIntent.putExtra(RingtonePlayingService.ALARM_ID, id);
        PendingIntent pendingIntentYes = PendingIntent.getService(this, id, yesIntent, PendingIntent.FLAG_UPDATE_CURRENT);
        actionYes = new NotificationCompat.Action(android.R.drawable.ic_lock_idle_alarm, "Dismiss", pendingIntentYes);

    }

    @TargetApi(Build.VERSION_CODES.O)
    private void createChannel() {
        NotificationChannel channel = new NotificationChannel(channelID, channelName, NotificationManager.IMPORTANCE_HIGH);
        getManager().createNotificationChannel(channel);

        long[] pattern = {0, 100, 1000, 300, 200, 100, 500, 200, 100};
        try {
            Uri notification = Uri.parse("android.resource://" + getPackageName() + "/" + R.raw.old_telephone);
            v = (Vibrator) getSystemService(Context.VIBRATOR_SERVICE);
            //ringtone = RingtoneManager.getRingtone(getApplicationContext(), notification);
            v.vibrate(VibrationEffect.createWaveform(pattern, 3));
          //  ringtone.play();
            //   v.vibrate(VibrationEffect.createOneShot(3000, VibrationEffect.DEFAULT_AMPLITUDE));

        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public NotificationManager getManager() {
        if (mManager == null) {
            mManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        }
        return mManager;
    }

    public NotificationCompat.Builder getChannelNotification() {
        NotificationCompat.Builder mBuilder =
                new NotificationCompat.Builder(getApplicationContext(), channelID)
                        .setSmallIcon(R.drawable.ic_launcher_background)
                        .setContentTitle(name)
                        .setContentText(temp)
                        .addAction(actionYes)
                        .setSmallIcon(R.drawable.ic_launcher_background);
        mBuilder.setVibrate(new long[]{500, 500});
        NotificationManager mNotifyMgr =
                (NotificationManager) this.getSystemService(this.NOTIFICATION_SERVICE);

        mNotifyMgr.notify(id, mBuilder.build());
        return mBuilder;
    }
}
      /*  return new NotificationCompat.Builder(getApplicationContext(), channelID)
                .setVibrate(new long[]{50, 100, 150, 200, 250})
                .setContentTitle(name)
                .setContentText("Alarm")
                .addAction(actionYes)
                .setSmallIcon(R.drawable.image);

    }*/

File: app/src/main/java/com/example/ahoyandroidtaskmudassar/utils/NotificationHelper.kt
package com.example.ahoyandroidtaskmudassar.utils

import android.content.ContextWrapper
import android.net.Uri
import android.content.ContentResolver
import com.example.ahoyandroidtaskmudassar.R
import androidx.core.app.NotificationCompat
import android.app.NotificationManager
import android.annotation.TargetApi
import android.os.Build
import android.app.NotificationChannel
import com.example.ahoyandroidtaskmudassar.utils.NotificationHelper
import android.os.Vibrator
import android.media.RingtoneManager
import android.os.VibrationEffect
import java.lang.Exception
import android.media.Ringtone
import android.content.Intent
import com.example.ahoyandroidtaskmudassar.utils.RingtonePlayingService
import android.app.PendingIntent
import android.content.Context

class NotificationHelper(base: Context, name: String, Id: Int) : ContextWrapper(base) {
    var soundUri =
        Uri.parse(ContentResolver.SCHEME_ANDROID_RESOURCE + "://" + applicationContext.packageName + "/" + R.raw.old_telephone)
    var actionYes: NotificationCompat.Action
    private var name = ""
    private var id = 0
    private var mManager: NotificationManager? = null
    var sound: Uri? = null
    @TargetApi(Build.VERSION_CODES.O)
    private fun createChannel() {
        val channel =
            NotificationChannel(channelID, channelName, NotificationManager.IMPORTANCE_HIGH)
        manager!!.createNotificationChannel(channel)
        val pattern = longArrayOf(0, 100, 1000, 300, 200, 100, 500, 200, 100)
        try {
            val notification =
                Uri.parse("android.resource://" + packageName + "/" + R.raw.old_telephone)
            v = getSystemService(VIBRATOR_SERVICE) as Vibrator
            ringtone = RingtoneManager.getRingtone(applicationContext, notification)
            v!!.vibrate(VibrationEffect.createWaveform(pattern, 3))
            ringtone.play()
            //   v.vibrate(VibrationEffect.createOneShot(3000, VibrationEffect.DEFAULT_AMPLITUDE));
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    val manager: NotificationManager?
        get() {
            if (mManager == null) {
                mManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
            }
            return mManager
        }

    /*  return new NotificationCompat.Builder(getApplicationContext(), channelID)
                .setVibrate(new long[]{50, 100, 150, 200, 250})
                .setContentTitle(name)
                .setContentText("Alarm")
                .addAction(actionYes)
                .setSmallIcon(R.drawable.image);

    }*/
    val channelNotification: NotificationCompat.Builder
        get() {
            val mBuilder = NotificationCompat.Builder(applicationContext, channelID)
                .setSmallIcon(R.mipmap.ic_launcher)
                .setContentTitle(name)
                .setContentText("Alarm")
                .addAction(actionYes)
                .setSmallIcon(R.mipmap.ic_launcher)
            mBuilder.setVibrate(longArrayOf(500, 500))
            val mNotifyMgr = this.getSystemService(NOTIFICATION_SERVICE) as NotificationManager
            mNotifyMgr.notify(id, mBuilder.build())
            return mBuilder
        }

    companion object {
        const val channelID = "channelID"
        const val channelName = "Channel Name"
        var r: Ringtone? = null
        lateinit var ringtone: Ringtone
        var v: Vibrator? = null
    }

    init {
        this.name = name
        id = Id
        sound =
            Uri.parse(ContentResolver.SCHEME_ANDROID_RESOURCE + "://" + base.packageName + "/" + R.raw.old_telephone)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            createChannel()
        }
        val yesIntent = Intent(this, RingtonePlayingService::class.java)
        yesIntent.putExtra(RingtonePlayingService.ALARM_ID, id)
        val pendingIntentYes =
            PendingIntent.getService(this, id, yesIntent, PendingIntent.FLAG_UPDATE_CURRENT)
        actionYes = NotificationCompat.Action(
            android.R.drawable.ic_lock_idle_alarm,
            "Dismiss",
            pendingIntentYes
        )
    }
}